src/main/java/com/autotune/common/datasource/DataSourceManager.java

```diff
@@ -28,12 +30,43 @@ public DataSourceManager() {
     /**
      * Imports Metadata for a specific data source using associated DataSourceInfo.
      */
-    public void importMetadataFromDataSource(DataSourceInfo dataSource) {
+    public void importMetadataFromDataSource(DataSourceInfo dataSourceInfo) {
         try {
-            if (null == dataSource) {
+            if (null == dataSourceInfo) {
                 throw new DataSourceDoesNotExist(KruizeConstants.DataSourceConstants.DataSourceErrorMsgs.MISSING_DATASOURCE_INFO);
             }
-            dataSourceMetadataOperator.createDataSourceMetadata(dataSource);
+            dataSourceMetadataOperator.createDataSourceMetadata(dataSourceInfo);
+
+            try {
+                DataSourceMetadataInfo dataSourceMetadataInfo = dataSourceMetadataOperator.getDataSourceMetadataInfo(dataSourceInfo);
+
+                if (null == dataSourceMetadataInfo) {
+                    LOGGER.debug(KruizeConstants.DataSourceConstants.DataSourceMetadataErrorMsgs.DATASOURCE_METADATA_INFO_NOT_AVAILABLE);
+                    return;
+                }
+                // save the metadata to DB
+                ValidationOutputData addedToDB = null;
+                for (DataSource dataSource : dataSourceMetadataInfo.getDataSourceHashMap().values()) {
+                    String dataSourceName = dataSource.getDataSourceName();
+                    // check if dataSource already exists in the DB and proceed to add accordingly
+                    if (!checkIfDataSourceExists(dataSourceName)) {
+                        try {
+                            // add the data source to DB
+                            addedToDB = new ExperimentDBService().addMetadataToDB(dataSourceMetadataInfo);
+                            if (addedToDB.isSuccess()) {
+                                LOGGER.info("Metadata added to the DB successfully.");
```

Comment:
Since it's adding metadata to the database, should the function name be changed to "saveMetaDataToDB," correct?

Reply:
`DataSourceManager` maps CRUD methods in `DataSourceMetadataOperator`
Should I create a function 'saveMetadataToDB' encapsulating DB operations?

Reply:
I have updated `saveMetadataToDB` with DB operations